helix/package-principles.rst

```diff
@@ -0,0 +1,137 @@
+specific Helix practices.  Note that the original material uses the term "component," whereas Helix uses the term "module." We will use
+"component" when referring to packages in the original pre-Sitecore context, and we will use "module" when referring to Helix.
+
+The Principles of Component Cohesion
+------------------------------------
+Martin's first three principles deal with what goes into a component, and what stays out.
+
+.. note::
+
+    **The Reuse/Release Equivalency Principle (REP)**
+
+    The granule of reuse is the granule of release.
+ 
+In Martin's discussion, this principle states that code that is meant to 
+be reused must be released according to a defined, predictable process, 
+and that the scope of the released package is governed by how it will 
+be put to use. He makes the point that one would not expect classes 
+that are not intended to be reused to be packaged with ones that are, and
+that the package should be reusable by the same audience: "Many people who would 
+like to reuse a container class library would have no interest in a financial
+framework."
+
+In Helix, this principle underpins the practice of grouping modules by 
+recognizable business needs, such as navigation or promotion display. Even 
+if such a module is not intended for reuse, it is intended for a specific
+audience, such as developers working on navigation. This principle lets them
+treat a module as a well-defined space in which only classes relevant to their
+development effort are present, limiting distraction, complexity and the 
+potential for unintended side effects. In Helix terms, this principle could
+be restated as, "The granule of business purpose is the granule of module size,"
+or more simply, "A module should serve a single business audience."
+
+.. note::
+
+    **The Common Reuse Principle (CRP)**
+
+    The classes in a component are reused together. If you reuse one of the
+    classes in a component, you reuse them all.
+
+In Martin's writing, this principle establishes that there are close relationships
+between the classes in a component: "classes that are not bound to each other with 
+class relationships should not be in the same component."  We see this principle at work
+in the treatment of items and templates in Helix. By defining items through the
+composition of interface templates, Helix hides from the module the fields the 
+module does not use. Developers familiar with Martin's SOLID principles will
+recognize this as an application of the Interface Segregation Principle: "Clients 
+should not be forced to depend on methods they do not use" (Martin, ch. 12).
+
+By using interface templates, and limiting item visibility to these fields, Helix 
+keeps out of the view of the module functionality of other modules, as well as all the
+fields Sitecore uses to manage items (icon, publishing settings, workflow, 
+security). This is why even with Data Source components, Helix prescribes a separation
```

Comment:
Can we say *recommends* instead of *prescribes*?

Reply:
```suggestion
security). This is why even with Data Source components, Helix recommends a separation
```